src/App.jsx

```diff
@@ -7,19 +7,35 @@ import database from './database.json';
 import './App.css';
 
 function App() {
+  const [productsList, setProductsList] = useState(database);
+
+  const upvoteProduct = (id) => {
+    const productListClone = [...productsList];
+    const elementIndex = productListClone.findIndex((el) => el.id === id);
```

Comment:
Find the product in question using the ID provided, currently we are not using any error handling. In the future it would be great to throw an error when the ID is not found.